compose/ui/ui/src/desktopMain/kotlin/androidx/compose/ui/awt/ComposeLayer.desktop.kt

```diff
@@ -384,16 +289,108 @@ internal class ComposeLayer(
 
     private var _initContent: (() -> Unit)? = null
 
-    private fun initContent() {
-        if (_component.isDisplayable) {
+    protected fun initContent() {
+        if (component.isDisplayable) {
             _initContent?.invoke()
             _initContent = null
         }
     }
+
+    private fun setCurrentKeyboardModifiers(modifiers: PointerKeyboardModifiers) {
+        platform.windowInfo.keyboardModifiers = modifiers
```

Comment:
I think there's a problem here. `platform.windowInfo.keyboardModifiers` needs to be set from a listener on events on the window, not just the component.

Reply:
That's a really good point! But let's fix it separately